web/app.js

```diff
@@ -1388,6 +1393,22 @@ const PDFViewerApplication = {
           case "error":
             console.error(detail.value);
             break;
+          case "print":
+            this.triggerPrinting();
+            return;
+          case "zoom":
+            if (typeof detail.value === "string") {
+              this.pdfViewer.currentScaleValue = detail.value;
+            } else {
+              this.pdfViewer.currentScale = detail.value;
+            }
+            return;
+          case "layout":
+            this.pdfViewer.spreadMode = apiPageLayoutToSpreadMode(detail.value);
+            return;
+          case "page-num":
+            this.pdfViewer.currentPageNumber = detail.value + 1;
+            return;
```

Comment:
Please make sure that you sort all of these `case`s *alphabetically*, since that will make it much easier to find things as the number of cases grow.